### ISSUE TITLE
RandomTransitionGenerator gets outside of image's content sometimes, or does too much zooming sometimes

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [flavioarfaria/KenBurnsView#4]
I'm trying to have a POC project that has the following features:
- fading action  bar, using this library:
https://github.com/flavienlaurent/NotBoringActionBar
- imageView that change their sizes in parallel (top and bottom) , as you scroll.
- the imageViews are actually KenBurnsView, and they switch between multiple images that might be from the Internet.
- all in fragments and action-bar-navigation is supported.

here's the project:
https://drive.google.com/file/d/0B_nrUhQ0MIgwUGZ3cjJwZVpVbkU/

the problem is that even though everything works , the RandomTransitionGenerator sometimes go outside of the content of the image being shown, so empty area is shown too. Also, sometimes, the zoom factor gets really large.
of course, I had to change some stuff on both libraries in order for this to work, but most of the code remained intact.

**Comment by AndreiHarasemiuc** [comment]
hello

I'm not having the problem with getting out of the image's bounds. 
However, I'm having the problem with WAY too much zoom.

You guys have any idea how I can fix that?

**Comment by AndroidDeveloperLB** [comment]
Sorry, I didn't return to this issue. I don't use this library anymore.

**Comment by Nomane** [comment]
Hello,

Some news about this récurrent issue, it seems to occurs when view is used with multiple image downloaded from URL (Picasso in my case) and dynamically resized

Thanks
Nomane

**Comment by flavioarfaria** [comment]
I haven't seen it happening after https://github.com/flavioarfaria/KenBurnsView/pull/34.
Closing for now.

**Comment by kishorecs** [comment]
There is another issue with Samsung galaxy S8..
I'll share the log


Fatal Exception: java.lang.RuntimeException: Canvas: trying to draw too large(124367104bytes) bitmap.
       at android.view.DisplayListCanvas.throwIfCannotDraw(DisplayListCanvas.java:229)
       at android.view.RecordingCanvas.drawBitmap(RecordingCanvas.java:97)
       at android.graphics.drawable.BitmapDrawable.draw(BitmapDrawable.java:529)
       at android.widget.ImageView.onDraw(ImageView.java:1367)
       at android.view.View.draw(View.java:20352)
       at android.view.View.updateDisplayListIfDirty(View.java:19297)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.support.constraint.ConstraintLayout.dispatchDraw(ConstraintLayout.java:2023)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.support.v4.widget.DrawerLayout.drawChild(DrawerLayout.java:1426)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.updateDisplayListIfDirty(View.java:19288)
       at android.view.View.draw(View.java:20075)
       at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
       at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
       at android.view.View.draw(View.java:20355)
       at com.android.internal.policy.DecorView.draw(DecorView.java:979)
       at android.view.View.updateDisplayListIfDirty(View.java:19297)
       at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:686)
       at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:692)
       at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:800)
       at android.view.ViewRootImpl.draw(ViewRootImpl.java:3488)
       at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3275)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2810)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1779)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7810)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
       at android.view.Choreographer.doCallbacks(Choreographer.java:723)
       at android.view.Choreographer.doFrame(Choreographer.java:658)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
       at android.os.Handler.handleCallback(Handler.java:789)
       at android.os.Handler.dispatchMessage(Handler.java:98)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6938)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
